src/decays/H_SM_decays/decay_AH_to_dqdq.cpp

```diff
@@ -0,0 +1,130 @@
+   // both with running and pole masses
+   const auto indices = concatenate(indexOut1, indexOut2, indexIn);
+   const auto HBBbarVertexDR = Vertex<bar<dq>::type, dq, AH>::evaluate(indices, context);
+
+   const double mAOS = context.physical_mass<AH>(indexIn);
+   const double flux = 1./(2.*mAOS);
+
+   constexpr double color_factor = squared_color_generator<AH, bar<dq>::type, dq>();
+
+   if(!boost::range::equal(indexOut1, indexOut2)) {
+      if (!is_zero(HBBbarVertexDR.left()) || !is_zero(HBBbarVertexDR.right())) {
+         const double mdqOS1 = context.physical_mass<dq>(indexOut1);
+         const double mdqOS2 = context.physical_mass<dq>(indexOut2);
+         const auto xOS1 = Sqr(mdqOS1/mAOS);
+         const auto xOS2 = Sqr(mdqOS2/mAOS);
+         const double phase_space = 1./(8.*Pi) * std::sqrt(KallenLambda(1., xOS1, xOS2));
+         return flux * phase_space * color_factor * amplitude_squared<AH, bar<dq>::type, dq>(context, indexIn, indexOut1, indexOut2);
+      }
+      return 0.;
+   }
+
+   const double mdqDR = context.mass<dq>(indexOut1);
+   const double mdqOS = context.physical_mass<dq>(indexOut1);
+   if(is_zero(mdqDR) || is_zero(mdqOS)) {
+      throw std::runtime_error("Error in A->ddbar: down quarks cannot be massless");
+   }
+   const auto xOS = Sqr(mdqOS/mAOS);
+   const auto xDR = Sqr(mdqDR/mAOS);
+
+   // TODO: add off-shell decays?
+   if (4.*std::max(xDR, xOS) > 1.) {
+      return 0.;
+   }
+
+   const auto betaOS = std::sqrt(1.-4.*xOS);
+   const auto betaDR = std::sqrt(1.-4.*xDR);
+
+   const double phase_spaceDR = 1./(8.*Pi) * std::sqrt(KallenLambda(1., xDR, xDR));
+   const double phase_spaceOS = 1./(8.*Pi) * std::sqrt(KallenLambda(1., xOS, xOS));
+
+   const auto HBBbarVertexDR_P = 0.5*(HBBbarVertexDR.right() - HBBbarVertexDR.left());
+
+   double amp2DR_P = 0;
+   double amp2OS_P = 0;
+      amp2DR_P = Sqr(mAOS) *
+                2*std::norm(HBBbarVertexDR_P);
+      amp2OS_P = Sqr(mAOS) *
+                2*std::norm(HBBbarVertexDR_P) * Sqr(mdqOS / mdqDR);
+
+   if (static_cast<int>(flexibledecay_settings.get(FlexibleDecay_settings::include_higher_order_corrections)) > 0) {
+         const int Nf = number_of_active_flavours(qedqcd, mAOS);
+         double alpha_s_red;
+         double Y_conversion = 1.;
+         switch (Nf) {
+            case 5: {
+               auto qedqcd_ = qedqcd;
+               qedqcd_.to(mAOS);
+               alpha_s_red = qedqcd_.displayAlpha(softsusy::ALPHAS)/Pi;
+               Y_conversion = Sqr(sm_down_quark_masses(qedqcd_, indexOut1.at(0))/mdqDR);
+               break;
+            }
+            case 6:
+               alpha_s_red = get_alphas(context)/Pi;
+               break;
+            default:
+               throw std::runtime_error("Error in A->ddbar: Cannot determine the number of active flavours");
+         }
+
+         double deltaqq_QCD_DR_P = calc_Deltaqq(alpha_s_red, Nf, flexibledecay_settings);
+
+         // 1L QED correction - eq. 21 in FD manual
```

Comment:
We must make sure that the eq. numbers match in the end.

Reply:
done in 3a8e84303cfdaaf3e6f5326adde63cd40a32fc31

Reply:
Thanks!